docs/maui/views/popup/popup-result.md

```diff
@@ -0,0 +1,178 @@
+    Padding="10"
+    Spacing="6"
+    x:TypeArguments="system:Boolean"
+    x:Class="MyProject.ReturnResultPopup">
+
+    <VerticalStackLayout>
+        <Label Text="This is a very important message! Do you agree?" />
+
+        <Button Text="Yes" 
+                Clicked="OnYesButtonClicked" />
+
+        <Button Text="No"
+                Clicked="OnNoButtonClicked" />
+    
+    </VerticalStackLayout>
+    
+</toolkit:Popup>
+```
+
+The use of `x:TypeArguments` in XAML makes it possible to supply the type parameter for a generic type.
+
+##### .xaml.cs
+
+```csharp
+public partial class ReturnResultPopup : Popup<bool>
+{
+    public ReturnResultPopup()
+    {
+        InitializeComponent();
+    }
+
+    async void OnYesButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(true);
+    }
+    
+    async void OnNoButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(false);
+    }
+}
+```
+
+The use of `Popup<bool>` must match the definition in the XAML through the use of `x:TypeArguments`.
+
+> [!IMPORTANT]
+> If the code behind file is not created along with the call to `InitializeComponent` then an exception will be thrown when trying to display your `Popup`.
+
+### C#
+
+The following section covers how to create a `Popup` using C#.
+
+```csharp
+using CommunityToolkit.Maui.Views;
+
+namespace MyProject;
+
+public class ReturnResultPopup : Popup<bool>
+{
+    public ReturnResultPopup()
+    {
+        var yesButton = new Button { Text = "Yes" };
+        yesButton.Clicked += OnYesButtonClicked;
+
+        var noButton = new Button { Text = "No" };
+        noButton.Clicked += OnNoButtonClicked;
+
+        Content = new VerticalStackLayout
+        {
+            HorizontalOptions = LayoutOptions.Center,
+            VerticalOptions = LayoutOptions.Center,
+            Children = 
+            {
+                new Label { Text = "This is a very important message! Do you agree?" },
+                
+                yesButton,
+                noButton
+            }
+        };
+    }
+
+    async void OnYesButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(true);
+    }
+    
+    async void OnNoButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(false);
+    }
+}
+```
+
+The `Close` method allows for a value to be supplied, this will be the resulting return value. The use of generics in `Popup<T>` provides type-safety when returning values from a Popup.
+
+## Awaiting the result from a Popup
+
+In order to await the result the `ShowPopupAsync` method must be used as follows:
+
+```csharp
+using CommunityToolkit.Maui.Views;
+
+public class MyPage : ContentPage
+{
+    public async Task DisplayPopup()
+    {
+        var popup = new ReturnResultPopup();
+
+        // The type parameter must match the type returned from the popup.
+        IPopupResult<bool> result = await this.ShowPopupAsync(popup, CancellationToken.None);
```

Comment:
Small nit (and you can totally ignore this) - I recommend naming this variable `popupResult` so that the `if` statement below becomes more readable, e.g. `if(popupResult.Result)`